src/components/pages/Map.tsx

```diff
@@ -64,12 +65,33 @@ function Map() {
   };
   // </infoWindowオプション-->
 
+  // <表示範囲判定>
+  const [neBounds, setNeBounds] = useState({lat: 0,lng: 0});
+  const [swBounds, setSwBounds] = useState({lat: 0,lng: 0});
+
+  const mapRef = React.useRef<google.maps.Map | undefined>();
+
+  const onMapLoad = React.useCallback((map: google.maps.Map) => {
+    mapRef.current = map;
+  }, []);
+
+  const onMapBoundsChanged = React.useCallback(() => {
+    const neLatlng = mapRef?.current?.getBounds()?.getNorthEast();
+    const swLatlng = mapRef?.current?.getBounds()?.getSouthWest();
+    setNeBounds({lat: neLatlng?.lat(), lng: neLatlng?.lng()});
+    setSwBounds({lat: swLatlng?.lat(), lng: swLatlng?.lng()});
```

Comment:
こんな感じで型エラーを解決できるかと思います！
```suggestion
    if (neLatlng?.lat() && neLatlng?.lng()) {
      setNeBounds({lat: neLatlng?.lat(), lng: neLatlng?.lng()});
    }
    if (swLatlng?.lat() && swLatlng?.lng()) {
      setSwBounds({lat: swLatlng?.lat(), lng: swLatlng?.lng()});
    }
```